crates/tabby-scheduler/src/index.rs

```diff
@@ -37,14 +38,24 @@ fn add_changed_documents(
         .writer(150_000_000)
         .expect("Failed to create index writer");
 
-    let total_file_size: usize = SourceFile::all()
-        .filter(is_valid_file)
-        .map(|x| x.read_file_size())
-        .sum();
-
-    let mut pb = std::io::stdout()
-        .is_terminal()
-        .then(|| tqdm(total_file_size));
+    let mut pb = std::io::stdout().is_terminal().then(|| {
+        let total_file_size = Walk::new(tabby_common::path::repositories_dir())
+            .filter_map(|entry| entry.ok())
+            .filter_map(|entry| {
+                let path = entry.path();
+                if !path.is_file() {
+                    return None;
+                }
+                get_language_by_ext(path.extension()?)?;
+                if cache.check_indexed(path).1 {
+                    None
+                } else {
+                    Some(())
+                }
+            })
+            .count();
+        tqdm(total_file_size)
+    });
```

Comment:
Let's get rid of terminal process (like dataset.rs) as it's no longer very meaningful as we're pushing user to web ui.